pkg/controller/endpointslicemirroring/utils.go

```diff
@@ -278,3 +290,35 @@ func hasLeaderElection(annotations map[string]string) bool {
 	_, ok := annotations[resourcelock.LeaderElectionRecordAnnotationKey]
 	return ok
 }
+
+// updateEndpointSliceLabels returns true if all labels in a are not present in b
+func updateEndpointSliceLabels(a, b map[string]string) bool {
```

Comment:
Nit: The names of these functions feel a bit off to me, almost like they will update the labels/annotations, not check if they should be updated. Maybe something like `labelsChanged` and `annotationsChanged` would be clearer here. Or alternatively, `labelsMatch` and `annotationsMatch`?